src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -315,14 +351,60 @@ private void updateCISShadowReportedState(Map<String, Object> reportedState) {
         });
     }
 
-    private void publishToGetCISShadowTopic() {
+    private void reportShadowReceived() {
+        CompletableFuture<?> shadowReceived = this.shadowGetResponseReceived.get();
+        if (shadowReceived != null) {
+            shadowReceived.complete(null);
+        }
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void fetchCISShadowWithRetriesAsync() {
+        synchronized (getShadowLock) {
+            if (getShadowTask != null && !getShadowTask.isDone()) {
```

Comment:
check for cancelled? or does done include cancelled

Reply:
done includes cancelled